index.js

```diff
@@ -1,35 +1,63 @@
 #!/usr/bin/env node
 require('dotenv').config();
+const gitUrlParse = require('git-url-parse');
 const program = require('commander');
+var childProcess = require('child_process');
+var colors = require('colors');
 var gitlab = require('gitlab');
-var exec = require('child_process').exec;
 var editor = require('editor');
+var exec = childProcess.exec;
+var execSync = childProcess.execSync;
 var fs = require('fs');
 var open = require("open");
+var projectDir = process.cwd();
 var Promise = require('promise');
 var URL = require('url');
-var colors = require('colors');
+var readlineSync = require('readline-sync');
 var regexParseProjectName = /(.+:\/\/.+?\/|.+:)(.+\/.+)+.git/;
 
-if (!process.env.GITLAB_URL) {
-  console.error(colors.red('Env variable GITLAB_URL is not set. Please set env variable GITLAB_URL'));
-  console.error('Eg GITLAB_URL=http://gitlab.yourcompany.com');
-  process.exit(1);
-}
-
-if (!process.env.GITLAB_TOKEN) {
-  console.error(colors.red('Please set env variable GITLAB_TOKEN'));
-  console.error('Find your token at ' + process.env.GITLAB_URL + '/profile/account');
-  process.exit(1);
-}
+var git = {
+  config: {
+    get: function (key) {
+      return execSync('git config --get ' + key + ' || true', { cwd: projectDir }).toString().trim()
+    },
+    set: function (key, value) {
+      execSync('git config --add ' + key + ' ' + value, { cwd: projectDir })
+    },
+  }
+};
+
+var gitlab = require('gitlab')((function () {
+  var options = {
+    url: git.config.get('gitlab.url') || process.env.GITLAB_URL || (function (remote) {
+      var url = git.config.get('remote.origin.url');
```

Comment:
This will not work when user is using `~/ssh/config` file with custom hostname. Eg `git@vishwanatharondekar:vishwanatharondekar/gitlab-cli.git` . Since we never asked them for gitlab url, they will never know what has happened and will create issues. 

Is there any way we can tell such users what to do.

Reply:
Is that really a thing people do?

Reply:
Well I use it almost all the times now. Multiple github accounts, gitlab accounts. Makes life a lot easier. What you are saying makes sense about `open-merge-requests`. Its broken right now for those users. But create-merge-request is working for them right now. I don't know if I want to break that as well. 

I don't think the trouble is worth it. This PR adds a lot of value compared to those issues. I will have to change that README as soon as possible with this gotcha. Merging this. Will take care of such users when they complain.

Reply:
Wait. All other commands actually use the GITLAB_URL value provided. So its not broken yet. This will break it. Still thinking about this.